File: react-native/react-native-kenkou-sdk/ios/ModelWithEmitter.m
#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(ModelWithEmitter, RCTEventEmitter)

RCT_EXTERN_METHOD(supportedEvents)

@end

File: react-native/react-native-kenkou-sdk/ios/ModelWithEmitter.swift
import Foundation

@objc(ModelWithEmitter)
open class ModelWithEmitter: RCTEventEmitter  {
    
    public static var emitter: RCTEventEmitter!
    
    override init() {
        super.init()
        ModelWithEmitter.emitter = self
    }
    
    /// Base overide for RCTEventEmitter.
    @objc open override func supportedEvents() -> [String] {
        return ["heartBeat"];
        // return EventEmitter.sharedInstance.allEvents;
    }
    
    @objc static func heartBeat(body: Any) {
        ModelWithEmitter.emitter.sendEvent(withName: "heartBeat", body: body)
    }

}
